libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/headlineUiConfigHelper.ts

```diff
@@ -138,15 +138,22 @@ export function isComparisonEnabled(insight: IInsightDefinition) {
     return primaryMeasure && secondaryMeasures?.length === 1;
 }
 
-function getDefaultComparisonProperties(insight: IInsightDefinition): IComparison {
-    const [secondaryMeasure] = insightSecondaryMeasures(insight);
-    const secondaryIsDerivedMeasure =
-        isPoPMeasure(secondaryMeasure) || isPreviousPeriodMeasure(secondaryMeasure);
-
-    return {
+export function getDefaultComparisonProperties(insight?: IInsightDefinition): IComparison {
+    const comparison: IComparison = {
         enabled: true,
-        calculationType: secondaryIsDerivedMeasure ? CalculationType.CHANGE : CalculationType.RATIO,
     };
+
+    if (insight) {
+        const [secondaryMeasure] = insightSecondaryMeasures(insight);
+        const secondaryIsDerivedMeasure =
+            isPoPMeasure(secondaryMeasure) || isPreviousPeriodMeasure(secondaryMeasure);
+
+        comparison.calculationType = secondaryIsDerivedMeasure
```

Comment:
The changing of the object leads to less readable code imho.

I would propose to change it to:

````
export function getDefaultComparisonProperties(insight?: IInsightDefinition): IComparison {
    if (insight) {
        const [secondaryMeasure] = insightSecondaryMeasures(insight);
        const secondaryIsDerivedMeasure =
            isPoPMeasure(secondaryMeasure) || isPreviousPeriodMeasure(secondaryMeasure);

        return {
            enabled: true, 
            calculationType = secondaryIsDerivedMeasure
                        ? CalculationType.CHANGE
                        : CalculationType.RATIO,
    }

    return { enabled: true } ;
}
````